debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/ClassSummaryHelper.java

```diff
@@ -125,36 +135,56 @@ private int getLevel(String key) {
 	 * alphabetically.
 	 * 
 	 * @author jeanpb
-	 * 
 	 */
-	private class StringCompare implements Comparator<String> {
+	private static class StringCompare implements Comparator<String> {
+
 		private final Map<String, Integer> preferredOrderMap;
 
 		public StringCompare(String[] preferredOrder) {
-			preferredOrderMap = new HashMap<String, Integer>();
+			preferredOrderMap = new HashMap<>();
 
 			for (int i = 0; i < preferredOrder.length; i++) {
-				preferredOrderMap.put(preferredOrder[i], i);
+				preferredOrderMap.put(preferredOrder[i], Integer.valueOf(i));
 			}
 		}
 
 		private int find(String key) {
-			if (preferredOrderMap.containsKey(key.trim())) {
-				return preferredOrderMap.get(key.trim());
+			Integer order = preferredOrderMap.get(key.trim());
+			if (order != null) {
+				return order.intValue();
 			}
 			return Integer.MAX_VALUE;
 		}
 
+		private static String firstSegment(String key) {
+			String segment = "";
+			int start = key.indexOf('/') + 1;
+
+			if (start > 0) {
+				int end = key.indexOf('/', start);
+
+				if (end < 0) {
+					segment = key.substring(start);
+				} else {
+					segment = key.substring(start, end);
+				}
+			}
```

Comment:
Seems if the index of '/' is not found an exception should be thrown rather than returning "".

Reply:
Updated to throw IllegalArgumentException if no slash is found.